### PR TITLE
Add Xuid in clientData

### PR DESCRIPTION
**Pull request by SenseiTarzan**
it will allow to have the xuid of the player in clientdata the key it's  `Waterdog_XUID`

### CODE REVIEW

File: src/main/java/dev/waterdog/player/HandshakeUtils.java [hunk 1]
```diff
@@ -130,16 +130,19 @@ public static HandshakeEntry processHandshake(BedrockSession session, LoginPacke
         ECPublicKey identityPublicKey = EncryptionUtils.generateKey(identityPublicKeyString);
         JWSObject clientJwt = JWSObject.parse(packet.getSkinData().toString());
         EncryptionUtils.verifyJwt(clientJwt, identityPublicKey);
-        JsonObject clientData = HandshakeUtils.parseClientData(clientJwt, session);
+        JsonObject clientData = HandshakeUtils.parseClientData(clientJwt, extraData, session);
         return new HandshakeEntry(identityPublicKey, clientData, extraData, xboxAuth, protocol);
     }
 
-    public static JsonObject parseClientData(JWSObject clientJwt, BedrockSession session) throws Exception {
+    public static JsonObject parseClientData(JWSObject clientJwt,JsonObject extraData, BedrockSession session) throws Exception {
```
Comment (TobiasG-DE) [inline src/main/java/dev/waterdog/player/HandshakeUtils.java:137]: Please reformat appropriately

Comment (Alemiz112) [inline src/main/java/dev/waterdog/player/HandshakeUtils.java:137]: Forgot the space after `,` here. Please add it 😄

**Review by TobiasG-DE — CHANGES_REQUESTED** [review]
The PR must be properly formatted using the IntelliJ default code style. Then we can merge it

File: src/main/java/dev/waterdog/player/HandshakeUtils.java [hunk 2]
```diff
@@ -130,16 +130,19 @@ public static HandshakeEntry processHandshake(BedrockSession session, LoginPacke
...
         JsonObject clientData = (JsonObject) JsonParser.parseString(clientJwt.getPayload().toString());
         ProxyConfig config = ProxyServer.getInstance().getConfiguration();
-        if (config.useLoginExtras() && config.isIpForward()) {
+        if (config.useLoginExtras()) {
             // Add waterdog attributes
+            clientData.addProperty("Waterdog_XUID", extraData.get("XUID").getAsString());
+            if(config.isIpForward()){
```
Comment (Alemiz112) [inline src/main/java/dev/waterdog/player/HandshakeUtils.java:143]: I think this option might be actually removed. Would be nice if you can do it.

**Review by Alemiz112 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/waterdog/player/HandshakeUtils.java (hunk 1 above)
Comment (SenseiTarzan) [reply src/main/java/dev/waterdog/player/HandshakeUtils.java:137]: OK

File: src/main/java/dev/waterdog/player/HandshakeUtils.java (hunk 2 above)
Comment (SenseiTarzan) [reply src/main/java/dev/waterdog/player/HandshakeUtils.java:143]: what exactly do you want to delete

Comment (SenseiTarzan) [reply src/main/java/dev/waterdog/player/HandshakeUtils.java:143]: ` if(config.isIpForward()){`

**Review by TobiasG-DE — COMMENTED** [review]
Looks good

**Review by Alemiz112 — APPROVED** [review]
Approved

**Comment by SenseiTarzan** [comment]
thank you

**Review by TobiasG-DE — APPROVED** [review]

**Pull request merged**